example/lib/ide/problems_panel/problems_panel.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:io';
+
+import 'package:example/ide/problems_panel/diagnostics.dart';
+import 'package:example/lsp_exploration/lsp/lsp_client.dart';
+import 'package:flutter/material.dart';
+
+class ProblemsPanel extends StatefulWidget {
+  const ProblemsPanel({
+    super.key,
+    required this.lspClient,
+    this.sourceFile,
+  });
+
+  final LspClient lspClient;
+  final File? sourceFile;
+
+  @override
+  State<ProblemsPanel> createState() => _ProblemsPanelState();
+}
+
+class _ProblemsPanelState extends State<ProblemsPanel> {
+  List<Diagnostic>? diagnostics;
+  String? uri;
```

Comment:
Please declare these properties private.

Reply:
Is this true even in a private class?

Reply:
Yeah. In general, if there isn't a reason to make a property public, we should make it private.